drivers/power_domain/power_domain_gpio.c

```diff
@@ -116,6 +118,8 @@ static int pd_gpio_init(const struct device *dev)
 		/* Device is unpowered */
```

Comment:
If regulators can now be powered on at boot the assumption here that a device is unpowered if it is on a power domain itself is no longer valid